react/features/recording/components/RecordingButton.web.js

```diff
@@ -0,0 +1,307 @@
+import AKInlineDialog from '@atlaskit/inline-dialog';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { ToggleStateless } from '@atlaskit/toggle';
+
+import { translate } from '../../base/i18n';
+import { isButtonEnabled, ToolbarButton } from '../../toolbox';
+import Recording from '../../../../modules/UI/recording/Recording';
+
+const DEFAULT_BUTTON_CONFIGURATION = {
+    buttonName: 'recording',
+    classNames: [ 'button', 'icon-recEnable' ],
+    enabled: true,
+    id: 'toolbar_button_record',
+    tooltipKey: 'recording.buttonTooltip'
+};
+
+/**
+ * TOFIX: Copy paste from VideoQualityButton, we need a base class that supports
+ * inline dialogs and does that position thing.
+ * @type {{bottom: string, left: string, right: string, top: string}}
+ */
+const TOOLTIP_TO_DIALOG_POSITION = {
+    bottom: 'bottom center',
+    left: 'left middle',
+    right: 'right middle',
+    top: 'top center'
+};
+
+/**
+ * React {@code Component} for displaying an inline dialog for changing receive
+ * video settings.
+ *
+ * @extends Component
+ */
+class RecordingButton extends Component {
+    /**
+     * {@code RecordingButton}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The redux store representation of the JitsiConference.
+         */
+        _conference: React.PropTypes.object,
+
+        /**
+         * Indicates if the recording button has been toggled.
+         */
+        _recordingButtonToggled: React.PropTypes.bool,
+
+        /**
+         * Indicates if the audio recording option should be enabled.
+         */
+        _recordingEnabled: React.PropTypes.bool,
+
+        /**
+         * Indicates if the transcription option should be enabled.
+         */
+        _transcriptionEnabled: React.PropTypes.bool,
+
+        /**
+         * Whether or not the button is visible, based on the visibility of the
+         * toolbar. Used to automatically hide the inline dialog if not visible.
+         */
+        _visible: React.PropTypes.bool,
+
+        /**
+         * Invoked to obtain translated string.
+         */
+        t: React.PropTypes.func,
+
+        /**
+         * From which side tooltips should display. Will be re-used for
+         * displaying the inline dialog for video quality adjustment.
+         */
+        tooltipPosition: React.PropTypes.string
```

Comment:
In general the pattern now is to declare PropTypes as a variable (or maybe bring in the PropTypes package) and use that instead due to the deprecation of React.PropTypes.